src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Objects/Attributes.jl

```diff
@@ -583,3 +583,45 @@ function relative_cotangent_module(X::AbsProjectiveScheme{<:Ring, <:MPolyQuoRing
   return cokernel(psi)
 end
 
+@doc raw"""
+    genus(X::AbsProjectiveScheme{<:Field}; algorithm::Symbol=:normalization)
+
+Given a projective curve `X` return the genus of `X`, i.e. the 
+integer `p_g = p_a - delta` where `p_a` is the arithmetic genus 
+and `delta` the delta-invariant of the curve.
+
+The `algorithm` keyword can be specified to 
+  - `:normalization` to go via the computation of a normalization
+  - `:primary_decomposition` to proceed with a primary decomposition
+Normalization is usually faster, but not always.
+"""
+function genus(X::AbsProjectiveScheme{<:Field}; algorithm::Symbol=:normalization)
+  @req dim(X) == 1 "scheme must be one-dimensional"
+  get_attribute!(X, :genus) do
+    I = defining_ideal(X)
+    singular_assure(I)
```

Comment:
There has been a lot of work to reduce the calls to `singular_assure` e.g. in https://github.com/oscar-system/Oscar.jl/pull/2646 by providing a better interface to singular data instead. Can you make use of something like this here?